tests/dragonfly/connection_test.py

```diff
@@ -0,0 +1,176 @@
+import asyncio
+import aioredis
+import async_timeout
+#from conftest import DATABASE_INDEX
+
+
+def verify_response(monitor_response: dict, key: str, value: str) -> bool:
+    if monitor_response is None:
+        return False
+    if monitor_response["db"] == 1 and monitor_response["client_type"] == "tcp":
+        return key in monitor_response["command"] and value in monitor_response["command"]
+    else:
+        return False
+
+
+async def monitor_cmd(mon: aioredis.client.Monitor, messages: dict):
+    success = None
+    async with mon as monitor:
+        try:
+            for key, value in messages.items():
+                try:
+                    async with async_timeout.timeout(1):
+                        response = await monitor.next_command()
+                        success = verify_response(
+                            response, key, value)
+                        if not success:
+                            return False, f"failed on the verification of the message {response} at {key}: {value}"
+                        await asyncio.sleep(0.01)
+                except asyncio.TimeoutError:
+                    pass
+            return True, "monitor is successfully done"
+        except Exception as e:
+            return False, f"stopping monitor on {e}"
+
+
+async def run_monitor(messages: dict, pool: aioredis.ConnectionPool):
+    cmd1 = aioredis.Redis(connection_pool=pool)
+    conn = aioredis.Redis(connection_pool=pool)
+    monitor = conn.monitor()
+    future = asyncio.create_task(monitor_cmd(monitor, messages))
+    success = True
+    for key, val in messages.items():
+        res = await cmd1.set(key, val)
+        if not res:
+            success = False
+            break
+        await asyncio.sleep(1)
+    await asyncio.sleep(1)
+    await future
+    status, message = future.result()
+    if status and success:
+        return True, "successfully completed all"
+    else:
+        return False, f"monitor result: {status}: {message}, set command success {success}"
+
+
+async def run_monitor_command(connection, messages):
```

Comment:
does it have to be async if you call await inside?

Reply:
you cannot await if the function is not marked "async" this is like in Rust for example and not like the coroutins in C++, a function is coroutine only if it marked async and you must make them awaitable only by making them coroutine.